src/common/util/include/openvino/util/node_util.hpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+
+namespace ov {
+namespace util {
+/**
+ * @brief Set name for both node and output tensor. Any other names will be overriden by a given single name
+ * @param node - node to rename
+ * @param name - new name
+ * @param output_idx - idx of output to rename
+ * @return - renamed node
+ */
+std::shared_ptr<ov::Node> set_name(std::shared_ptr<ov::Node> node, const std::string& name, size_t output_idx = 0);
+
+// Templated method that has the same effect as not templated `set_name` but saves Op type for convenient calls chaining
+template <typename T>
+std::shared_ptr<T> set_name(std::shared_ptr<T> node, const std::string& name, size_t output_idx = 0) {
```

Comment:
What is benefit of this function?
Maybe would be better to have factory function instead as current context of usage suggest it e.g.:
```cpp
template <class TNode, class... Args>
std::shared_ptr<TNode> make_node_with_name(const std::string& name, Args&&... args){
    auto node = std::make_shared<TNode>(std::forward<Args>()...);
    // which port should be used one or many etc...
    util::set_name(*node, name, port_ids);
    return node;
};
```
Could be used like:
```cpp
auto param = ov::util::make_node_with_name<Parameter>("my_name", element::f32, PartialShape{3,2})
```

Reply:
make_node_with_name is a good idea but I think it is overkill right now